src/west/app/project.py

```diff
@@ -1602,6 +1602,204 @@ def do_run(self, args, user_args):
+
+  west grep --this-option-is-passed-to-the-tool
+
+To force an argument to be given to the tool instead of west, put
+it after a "--", like this:
+
+  west grep -- --project=val
+
+COLORS
+------
+
+By default, west will force the tool to print colored output as long
+as the "color.ui" configuration option is true. If color.ui is false,
+west forces the tool not to print colored output.
+
+Since all supported tools have similar --color options, you can
+override this behavior on the command line, for example with:
+
+  west grep --color=never
+
+To do this permanently, set "grep.color":
+
+  west config grep.color never
+'''
+
+class Grep(_ProjectCommand):
+
+    TOOLS = ['git-grep', 'ripgrep', 'grep']
+
+    DEFAULT_TOOL = 'git-grep'
+
+    DEFAULT_TOOL_ARGS = {
+        'git-grep': [],
+        'ripgrep': [],
+        'grep': ['--recursive'],
+    }
+
+    DEFAULT_TOOL_EXECUTABLES = {
+        # git-grep is intentionally omitted: use self._git
+        'ripgrep': ['rg', 'ripgrep'],
+        'grep': ['grep'],
+    }
+
+    def __init__(self):
+        super().__init__(
+            'grep',
+            'run grep or a grep-like tool in one or more local projects',
+            'Run grep or a grep-like tool in one or more local projects.',
+            accepts_unknown_args=True)
+
+    def do_add_parser(self, parser_adder):
+        parser = self._parser(parser_adder, epilog=GREP_EPILOG)
+        parser.add_argument('--tool', choices=self.TOOLS, help='grep tool')
+        parser.add_argument('--tool-path', help='path to grep tool executable')
+        # Unlike other project commands, we don't take the project as
+        # a positional argument. This inconsistency makes the usual
+        # use case of "search the entire workspace" faster to type.
+        parser.add_argument('-p', '--project', metavar='PROJECT',
+                            dest='projects', default=[], action='append',
+                            help='''project to run grep tool in (may be given
+                            more than once); default is all cloned active
+                            projects''')
+        return parser
+
+    def do_run(self, args, tool_cmdline_args):
+        tool = self.tool(args)
+        command_list = ([self.tool_path(tool, args)] +
+                        self.tool_args(tool, tool_cmdline_args))
+        failed = []
+        for project in self._cloned_projects(args,
+                                             only_active=not args.projects):
+            try:
+                output = self.check_output(command_list, cwd=project.abspath)
+            except subprocess.CalledProcessError as e:
+                # By default, supported tools exit 1 if nothing is
+                # found. Other return codes indicate error.
+                #
+                # This changes if you give the --quiet option, but
+                # that doesn't make much sense in this context, so
+                # assume it hasn't been passed.
```

Comment:
How about an assert or warning if `--quiet` has been passed? Assuming it takes only 1 or 2 lines of code.

Reply:
> How about an assert or warning if `--quiet` has been passed?

I gave this some more thought and revised the approach around --quiet a bit.